dask_deltatable/utils.py

```diff
@@ -1,6 +1,14 @@
 from __future__ import annotations
```

Comment:
Is there a more descriptive filename we could use other than `utils.py`?

Reply:
I guess it was renamed `utils.py` because there was no descriptive filename for those functions?

Reply:
It's unfortunate to add more functions in there, it would be better to refactor things into better namespaces. I don't want to block this PR on doing that though.

Reply:
Naming is one of the unsolved problems in computer science. It's more of an art. Hence, I don't have any opinion here. Please refactor things as needed.